ns1/provider.go

```diff
@@ -1,6 +1,10 @@
 package ns1
 
 import (
+	"errors"
+	"os"
+
+	"github.com/fatih/structs"
```

Comment:
How badly do we need this? Cause it does pull in reflect. And as far as I can tell, it's to add the `json` tag?

Reply:
Yeah but I use reflection pretty liberally throughout, there's no way to avoid that with this kind of thing. I used that to map the non nested meta object that's at the top level of the record resource. It preserves types accurately at the top level, but not when they're further nested. If this isn't a sufficient explanation I can walk you through it - and I understand if it isn't, it took me a while to grok the reasons why some of this is as it is

Reply:
Alright, that makes sense.